.travis.yml

```diff
@@ -1,6 +1,23 @@
 language: node_js
 node_js:
 - 10
+services:
+  - docker
 install: npm install
-script:
-- npm run lint
+
+stages:
+  - test
+  - build
```

Comment:
If we don't deploy the application why to use build?
We should use build when we want to deploy our image the next moment, don't you agree with me?
We should use the image-build with the deploy stage

Reply:
I think it's better to run the build anyway, to make sure that it is successful.
If you make it run only before deployment, the image build may break between deployments without us knowing about it.

Reply:
Ok, Then we could use tags very smart here.
When we deploy we can use tags like v1,v2,v3
When we just build we can use tags like v1.1 v1.2, v1.3

Or even make it this format:
build -> v1.01, v1.02v .1.03
deploy integration and testing -> v1.1, v1.2, v1.3
deploy for prod -> v1,v2,v3

Just some suggestions I never did this before but we will need something like that.